terracotta/cog.py

```diff
@@ -51,7 +51,10 @@ def check_raster_file(src_path: str) -> ValidationInfo:  # pragma: no cover
 
             overviews = src.overviews(1)
             if src.width > 512 and src.height > 512:
-                if not src.is_tiled:
+                if not (
+                    src.block_shapes
+                    and all(src.width != w for _, w in src.block_shapes)
+                ):
```

Comment:
Please add a function `is_tiled` instead, and indicate where this code comes from (copy-paste from rasterio?)

Reply:
done :heavy_check_mark: - it's now a separate function with the source etc.